src/NSwag.AspNetCore/IDocumentProvider.cs

```diff
@@ -0,0 +1,20 @@
+﻿//-----------------------------------------------------------------------
+// <copyright file="IDocumentProvider.cs" company="NSwag">
+//     Copyright (c) Rico Suter. All rights reserved.
+// </copyright>
+// <license>https://github.com/NSwag/NSwag/blob/master/LICENSE.md</license>
+// <author>Rico Suter, mail@rsuter.com</author>
+//-----------------------------------------------------------------------
+
+using System.IO;
+using System.Threading.Tasks;
+
+namespace Microsoft.Extensions.ApiDescription
+{
+    // This service will be looked up by name from the service collection when using
+    // the Microsoft.Extensions.ApiDescription tool
+    internal interface IDocumentProvider
+    {
+        Task GenerateAsync(string documentName, TextWriter writer);
```

Comment:
How do you know what document to use when you can't enumerate it? Does it have to be specified in the csproj?

Reply:
The tool in the `Microsoft.Extensions.ApiDescription.Design` package relies on the user including `DocumentName` metadata on the relevant items in their `.csproj` file. If the metadata doesn't exist, the tool falls back to the name `"v1"`. If that doesn't match a document, it simply fails.

Reply:
Sounds good.